# DP: Comment out constant exposed on the API which are not implemented on
# DP: GNU/Hurd. They would not work at runtime anyway.

Index: b/Modules/posixmodule.c
===================================================================
--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -9248,12 +9248,14 @@ all_ins(PyObject *d)
 #ifdef O_LARGEFILE
     if (ins(d, "O_LARGEFILE", (long)O_LARGEFILE)) return -1;
 #endif
+#ifndef __GNU__
 #ifdef O_SHLOCK
     if (ins(d, "O_SHLOCK", (long)O_SHLOCK)) return -1;
 #endif
 #ifdef O_EXLOCK
     if (ins(d, "O_EXLOCK", (long)O_EXLOCK)) return -1;
 #endif
+#endif
 
 /* MS Windows */
 #ifdef O_NOINHERIT
Index: b/Modules/socketmodule.c
===================================================================
--- a/Modules/socketmodule.c
+++ b/Modules/socketmodule.c
@@ -4903,9 +4903,11 @@ init_socket(void)
 #ifdef  SO_OOBINLINE
     PyModule_AddIntConstant(m, "SO_OOBINLINE", SO_OOBINLINE);
 #endif
+#ifndef __GNU__
 #ifdef  SO_REUSEPORT
     PyModule_AddIntConstant(m, "SO_REUSEPORT", SO_REUSEPORT);
 #endif
+#endif
 #ifdef  SO_SNDBUF
     PyModule_AddIntConstant(m, "SO_SNDBUF", SO_SNDBUF);
 #endif
